### ISSUE TITLE
Proposal: templ website and playground

### ISSUE THREAD

**Issue by a-h** [a-h/templ#42]
It might be useful to have a templ playground so that people can share snippets of code and try out using templ without having to install it first.

It would need to have the following features:

* A place to write Go code for the input struct definition, input data, and any helper functions, e.g. date formatting.
* A place to write a templ file, which will take the `input`, e.g.:

```
{% templ Display(in Input) %}
{% endtempl %}
```

* Output areas to display the generated HTML code, and the HTML as displayed in the browser.
* A "Run" button to click to format the code and execute it.
  * Since it's basically going to execute arbitrary code, it would need to run in a Lambda or similar to prevent abuse.
  * Running in a Lambda would require the `templ` executable to be installed in the Lambda.
* A "Share" button that creates a shareable web link.
  * This supposes the idea of a backing database, e.g. DynamoDB.

**Comment by joerdav** [comment]
The documentation website is probably handy now that the syntax is growing. 

I have some experience with Hugo for this kind of thing so I can start a skeleton. If not I am happy to contribute some documentation.

**Comment by a-h** [comment]
Thanks, very timely! Writing a really good docs site is my next priority.

https://www.docsy.dev/docs/examples/ looks pretty good as a Hugo template.

Would love some help on that so I can focus on writing content. 😀

**Comment by joerdav** [comment]
I like this outline, very comprehensive! One thing I would say though is that "Creating your first component" comes too late. I think people trying templ for the first time will want to be creating their first component (copy paste style) straight after installation.

Atleast that's my experience with getting started guides so far! Give them a way to have something that works then begin explaining in more detail.

**Comment by a-h** [comment]
I've completed the server-side rendering section, and added examples including a counter. I've added a blog example for static generation.

Next, I need to write:

[ ] "Deploying static files"
[ ] "Hosting on AWS Lambda"
[ ] "Hosting on Docker"
[ ] "Storybook Integration"

I would also like to add something about using 3rd party CSS libraries, then I think the first pass is complete, and I'll publicise the docs, and see what feedback I get.
